mesonbuild/dependencies/dev.py

```diff
@@ -530,6 +547,40 @@ def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
         java_home_include = self.java_home / 'include'
         self.compile_args.append(f'-I{java_home_include}')
         self.compile_args.append(f'-I{java_home_include / platform_include_dir}')
+
+        if modules:
+            java_home_lib = self.java_home / 'lib'
+
+            # Location of libraries is different in JDK < 9. Cannot confirm if
+            # this logic works for anything < 1.8, but might as well try.
+            if version_compare(self.version, '<= 1.8'):
+                # The JDK and Meson have a disagreement here, so translate it
+                # over. In the event more translation needs to be done, add to
+                # following dict.
+                def cpu_translate(cpu: str) -> str:
+                    java_cpus = {
+                        'x86_64': 'amd64'
+                    }
+
+                    return java_cpus.get(cpu, cpu)
+
+                if len(modules) > 0:
+                    self.link_args.append(f'-L{java_home_lib / "jre" / cpu_translate(m.cpu_family)}')
```

Comment:
This is kind of a really complicated expression to put in an f-string. :(

Maybe cpu_translate could be redesigned to calculate the correct directory and 

```python
return java_home_lib / 'jre' / java_cpus.get(cpu, cpu)
```

Reply:
I'll  just create a local variable then.